File: arrays/maximum-unsorted-subarray.cpp
vector<int> Solution::subUnsort(vector<int> &A) {
    int n = A.size();
    int start=-1, end=-1;

    int temp = A[0];
    for(int i = 1 ; i<n ; i++){
        if(A[i] < temp) end = i;
        temp = max(A[i], temp);
    }
    
    temp = A[n-1];
    for(int i = n-1 ; i>=0 ; i--){
        if(A[i] > temp) start = i;
        temp = min(A[i], temp);
    }
    
    if(start==-1) return vector<int> {-1};
    
    return vector<int> {start,end};
}

File: arrays/maximum-unsorted-subarray.py
class Solution:
    def subUnsort(self, A):
        L = sys.maxsize; R = -sys.maxsize
        for i in range(len(A)-1):
            if A[i] > A[i+1]:
                L = min(L, A[i+1])
                R = max(R, A[i])

        l = -1; r = -1
        for i in range(len(A)):
            if A[i] > L:
                l = i
                break

        for i in range(len(A)-1, -1, -1):
            if A[i] < R:
                r = i
                break

        if l==-1 or r==-1: return [-1]
        return [l, r]

